src/libraries/System.Linq.Queryable/tests/Queryable.cs

```diff
@@ -77,6 +78,39 @@ public static void QueryableOfQueryable()
             Assert.Equal(2, i);
         }
 
+        /// <summary>
+        /// Verifies that all the Queryable methods contain a DynamicDependency
+        /// to the corresponding Enumerable method. This ensures the ILLinker will
+        /// preserve the corresponding Enumerable method when trimming.
+        /// </summary>
+        [Fact]
+        public static void QueryableMethodsContainCorrectDynamicDependency()
+        {
+            IEnumerable<MethodInfo> dependentMethods =
+                typeof(Queryable)
+                    .GetMethods(BindingFlags.Public | BindingFlags.Static)
+                    .Where(m => m.Name != "AsQueryable");
+
+            foreach (MethodInfo method in dependentMethods)
+            {
+                DynamicDependencyAttribute dependency = method.GetCustomAttribute<DynamicDependencyAttribute>();
```

Comment:
```C#
Assert.NotNull(dependency);
```
?

Reply:
I didn't think it was necessary since the next line would NRE if it was null. Do we typically add `Assert.NonNull`s even if we dereference the variable?

Reply:
It does seem more error-prone, but if the signatures would match up such that the unit test you have would find mismatches, then it would be a nice touch.  Are there any where the signatures don't match?

Reply:
We're inconsistent. Personally I prefer it in a case like this as not finding the delegate will make it null, the main condition we care about here, and asserting it rather than null ref'ing is likely to make it more diagnosable.